Formula/duktape.rb

```diff
@@ -0,0 +1,38 @@
+class Duktape < Formula
+  desc "Embeddable Javascript engine with compact footprint"
+  homepage "http://duktape.org"
+  url "http://duktape.org/duktape-2.2.1.tar.xz"
+  sha256 "3abe2eed2553305262b892c98f550bb1a94cf4fd73b51dc5c176fe08e7ade7f2"
+
+  def install
+    inreplace "Makefile.sharedlibrary" do |s|
+      s.gsub! "-soname", "-install_name"
+      s.gsub! %r{\/usr\/local}, prefix
+      s.gsub! "libduktape.so.$(REAL_VERSION)", "libduktape.$(REAL_VERSION).so"
+      s.gsub! "libduktaped.so.$(REAL_VERSION)", "libduktaped.$(REAL_VERSION).so"
+      s.gsub! "libduktape.so.$(SONAME_VERSION)", "libduktape.$(SONAME_VERSION).so"
+      s.gsub! "libduktaped.so.$(SONAME_VERSION)", "libduktaped.$(SONAME_VERSION).so"
+    end
+    system "make", "-f", "Makefile.sharedlibrary"
+    mkdir lib
```

Comment:
There should be no need for `mkdir lib` or `mkdir include`. Those directories should be created by the `make install` step. If `make install` breaks without them, I'd consider that a bug in upstream's installation process that we should report to them.

Reply:
I tested this myself with upstream's head of master. Worked fine for me without the `mkdir` steps.

Reply:
I didnt have the same luck.  When removing these steps.  I get a **Error 64** which means the directories don't exist 

```
==> Downloading http://duktape.org/duktape-2.2.1.tar.xz
######################################################################## 100.0%
==> make -f Makefile.sharedlibrary
==> make -f Makefile.sharedlibrary install
Last 15 lines from /<SNIP>/Homebrew/duktape/02.make:
2018-06-03 09:25:00 -0400

make
-f
Makefile.sharedlibrary
install

cp libduktape.202.20201.so libduktaped.202.20201.so /usr/local/Cellar/duktape/2.2.1/lib/
usage: cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file target_file
       cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file ... target_directory
make: *** [install] Error 64
```

Perhaps I missed something that made it work for you @apjanke ??

Any idea?

Reply:
Okay, this is broken for me too. Maybe the recent-ish [upstream change that does those mkdirs](https://github.com/svaarala/duktape/commit/98c14aea4c3acdbfe1f2d5c3ce92d4f136013742#diff-7eae2846800f097c2b9c87d56a123f73) didn't make it in to the most recent release?

Just leave 'em in there, then.

Reply:
Yea, apologies for my lack of experience with homebrew...I thought it'd be better to use latest release rather than `master`...but looks like `master` may improve portability in this case.

I'll work on this.

Reply:
No, you were correct: we only want releases, not `master`, in Homebrew. If you include a reference to `master`, it should be as a `head` stanza in addition to the main versioned formula.

I just tested against their `master` because it was easy to do. That was a mistake on my part.